File: gcd.py
# Uses python3
# Find the greatest common denominator using Euclid's Algorithm
def gcd(a, b):
    tempa = max(a,b) #Take the larger of the two
    tempb = min(a,b) #And smaller
    while tempb != 0: 
        a = tempa%tempb #Find remainder
        tempa = tempb #Make tempb the larger number
        tempb = a #Make the remainder the smaller number
    return tempa

a, b = [int(i) for i in input().split()] #Take two numbers
print(gcd(a, b))

File: gcd.cpp
#include <iostream>
#include<algorithm>

//Find the greatest common denominator by Euclid's
//algorithm. Very similar to code in python, see gcd.python
//for notes
int gcd(int a, int b) {
  int tempa = std::max(a,b);
  int tempb = std::min(a,b);
  while(tempb !=  0){
      a =tempa%tempb; 
      tempa = tempb;
      tempb = a;
  }
  
  return tempa;
}

int main() {
  int a, b;
  std::cin >> a >> b;
  std::cout << gcd(a, b) << std::endl;
  return 0;
}
